### PR TITLE
CDAP-15864. Google Drive Sink. Updated auth configs.

### PR DESCRIPTION
**Pull request by dvitiiuk**
JIRA: https://issues.cask.co/browse/CDAP-15864
DOC: https://wiki.cask.co/display/CE/Google+Drive+plugins

Added ability to authenticate by service account file.

### CODE REVIEW

File: src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java [hunk 1]
```diff
@@ -123,23 +116,18 @@ public String getReferenceName() {
...
-
-  public String getClientSecret() {
-    return clientSecret;
-  }
-
-  public String getRefreshToken() {
-    return refreshToken;
+  public String getAccountFilePath() {
+    if (AUTO_DETECT_VALUE.equals(accountFilePath)) {
+      accountFilePath = System.getenv(GOOGLE_APPLICATION_CREDENTIALS_ENV_PROP);
```
Comment (aonischuk) [inline src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java]: I think in my case the library automatically picked up data from the environment variable.
https://github.com/data-integrations/nlp/blob/5c61fb0fc17f010135feaaea4702a114a2a94bc1/nlp-common/src/main/java/io/cdap/google/common/NLPMethodExecutor.java#L81
In your case it does not work like that?

File: src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java [hunk 2]
```diff
@@ -123,23 +116,18 @@ public String getReferenceName() {
...
+    }
+    if (!new File(accountFilePath).exists()) {
+      throw new RuntimeException(String.format("There is no account file from neither \"Account file path\" property " +
```
Comment (aonischuk) [inline src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java]: We need a validation for this.

File: src/main/java/io/cdap/plugin/google/common/GoogleDriveClient.java [hunk 3]
```diff
@@ -45,32 +48,28 @@ public GoogleDriveClient(C config) {
...
     // start of workaround
-    String clientId = config.getClientId();
-    String clientSecret = config.getClientSecret();
-    String refreshToken = config.getRefreshToken();
+    try {
+      credential = GoogleCredential.fromStream(new FileInputStream(config.getAccountFilePath()))
+        .createScoped(Collections.singleton(getRequiredScope()));
+      return credential;
+    } catch (IOException e) {
+      throw new RuntimeException("There was issue during loading of account file.", e);
```
Comment (aonischuk) [inline src/main/java/io/cdap/plugin/google/common/GoogleDriveClient.java]: while loading account file.

**Review by aonischuk — CHANGES_REQUESTED** [review]

File: src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java (hunk 2 above)
Comment (dvitiiuk) [reply src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java]: It is validated implicitly during GoogleDriveClient creation in validate(FailureCollector collector). Should I add separate validate method anyway?

File: src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java (hunk 1 above)
Comment (dvitiiuk) [reply src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java]: You are right, by another way, but it is already implemented, will change it

Comment (dvitiiuk) [reply src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java]: done

File: src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java (hunk 2 above)
Comment (dvitiiuk) [reply src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java]: done

File: src/main/java/io/cdap/plugin/google/common/GoogleDriveClient.java (hunk 3 above)
Comment (dvitiiuk) [reply src/main/java/io/cdap/plugin/google/common/GoogleDriveClient.java]: done

**Review by aonischuk — APPROVED** [review]

**Pull request merged**